pkg/templates/nstemplatetiers/nstemplatetier_generator.go

```diff
@@ -52,3 +40,47 @@ func CreateOrUpdateResources(ctx context.Context, s *runtime.Scheme, client runt
 		return applyCl.ApplyObject(ctx, toEnsure, commonclient.ForceUpdate(true))
 	}, namespace, metadata, files)
 }
+
+// LoadFiles takes the file from deploy/nstemplatetiers/<tiername>/<yaml file name> . the folder structure should be 4 steps .
+// as the cologic here is written accordingly
+func LoadFiles(nsTemplateTiers embed.FS, root string) (metadata map[string]string, files map[string][]byte, err error) {
+	// load templates from assets
+	metadataContent, err := nsTemplateTiers.ReadFile(root + "/metadata.yaml")
+	if err != nil {
+		return nil, nil, errors.Wrapf(err, "unable to load templates")
+	}
+
+	metadata = make(map[string]string)
+	err = yaml.Unmarshal(metadataContent, &metadata)
+	if err != nil {
+		return nil, nil, errors.Wrapf(err, "unable to load templates")
+	}
+
+	paths, err := templates.GetAllFileNames(&nsTemplateTiers, root)
+	if err != nil {
+		return nil, nil, err
+	}
+	if len(paths) == 0 {
+		return nil, nil, fmt.Errorf("could not find any ns templates")
+	}
+	files = make(map[string][]byte)
+	for _, name := range paths {
+		if name == root+"/metadata.yaml" {
```

Comment:
I would say that using `filepath.Join` is the best practice for constructing the paths
```suggestion
		if name == filepath.Join(root, "metadata.yaml") {
```